hook.py

```diff
@@ -1,17 +1,35 @@
 import logging
+import asyncio
 from pathlib import Path
 from subprocess import Popen
+from ssl import get_server_certificate
 
 name = 'SSL'
 description = 'Run an SSL proxy in front of the server'
-address = None
+address = ''
+
+async def _check_cert():
+    await asyncio.sleep(5)
+    default_cert = ''
+    server_cert = get_server_certificate(('127.0.0.1', 8443))
+    with open(Path('plugins/ssl/conf/insecure_certificate.pem'), 'r') as f:
+        for line in f:
+            if 'PRIVATE KEY' in line:
+                break
+            default_cert += line
+
+    if server_cert == default_cert:
+        logging.warn('Insecure SSL private key and certificate in use. Consider generating and using your own '
+                    'to improve security. Documentation found '
+                    'here: https://github.com/mitre/caldera/wiki/Plugin:-ssl')
 
 
 async def enable(services):
-    Popen(['haproxy', '-q', '-f', 'plugins/ssl/templates/haproxy.conf'])
-    logging.debug('Serving at https://127.0.0.1:8443')
-    cert = Path("plugins/ssl/conf/insecure_certificate.pem")
-    if cert.is_file():
-        logging.warn('Found insecure ssl private key and certificate. Consider generating your own and removing'
-                     ' the insecure certificate from the conf directory to improve security. Documentation found '
-                     'here: https://github.com/mitre/caldera/wiki/Plugin:-ssl')
+    haproxy_conf = Path('plugins/ssl/templates/haproxy.conf')
+    user_conf = Path('plugins/ssl/conf/haproxy.conf')
```

Comment:
use os.path for the file paths, be for cross-compatibility

Reply:
think pathlib would have been OK, but change already made, so I'm resolving this.